### ISSUE TITLE
k-Fold Cross Validation

### ISSUE THREAD

**Issue by Moses-Ian** [Moses-Ian/IanNet#6]
https://machinelearningmastery.com/k-fold-cross-validation/

Shuffle the dataset randomly.
1. Split the dataset into k groups
2. For each unique group:
    a. Take the group as a hold out or test data set
    b. Take the remaining groups as a training data set
    c. Fit a model on the training set and evaluate it on the test set
    d. Retain the evaluation score and discard the model
3. Summarize the skill of the model using the sample of model evaluation scores

So, you're creating and training the model 5 times and seeing the average of the scores to get a better sense of the model's performance overall. When you're ready to publish the model, you will train it in the same way on all available data. This means you've slightly under-estimated the performance.
